src/components/style-selector/style-selector.vue

```diff
@@ -0,0 +1,75 @@
+<script setup lang="ts">
+import { ref, computed, watch } from 'vue'
+import { storeToRefs } from 'pinia'
+import { useTranslation } from 'i18next-vue'
+
+import SimpleStyleSelector from '@/components/style-selector/simple-style-selector.vue'
+import MediumStyleSelector from '@/components/style-selector/medium-style-selector.vue'
+import { useAppStore } from '@/stores/app.store'
+import { useMapStore } from '@/stores/map.store'
+import { useStyleStore } from '@/stores/style.store'
+import useMvtStyles from '@/composables/mvt-styles/mvt-styles.composable'
+
+const { t } = useTranslation()
+const mapStore = useMapStore()
+const appStore = useAppStore()
+const styleStore = useStyleStore()
+const { bgLayer } = storeToRefs(mapStore)
+const styles = useMvtStyles()
+
+const styleCapabilities = computed(() =>
+  styles.getStyleCapabilitiesFromLayer(bgLayer.value)
+)
+
+watch(bgLayer, bgLayer => {
+  if (!styles.isLayerStyleEditable(bgLayer)) {
+    appStore.toggleStyleEditorPanel(false)
+  }
+})
+
+let isSimpleStyleOpen = ref(false)
+let isMediumStyleOpen = ref(false)
+let isAdvancedStyleOpen = ref(false)
```

Comment:
These three states should depend on each other (or be part of one state with three values), as they cannot be open simultaneously on prod. This might also be handled when implementing the UI.

Reply:
for testing, I find it quite practical when different editors can be opened simultaneously (as the effects of editing a style are visible in the preset styles. We will have to work on the UI, so easiest will be to fix it there later.

Reply:
ok